src/search.c

```diff
@@ -0,0 +1,108 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+#include <stddef.h>
+#include <simclist.h>
+#include "search.h"
+#include "match.h"
+
+int find_match(char* line, char* word, int pos_start, int lineNum, list_t* matches) {
+    if (strlen(word) > strlen(line)) return -1;
+
+    char* dup = strdup(line);
+
+    char* line2 = strdup(line);
+    int pos = pos_start;
+    char* token = strtok(line2, " ,.!?\r\t\n");
+    match* foundMatch = NULL;
+
+    while (token != NULL) {
+
+        if (strncmp(token, word, strlen(word)) == 0) {
+
+            /* Config match */
+            foundMatch = match_new(token, lineNum, pos, dup);
+            match_append(foundMatch, matches);
+
+            /* Clean-up */
+            free(dup);
+
+            /* Return */
+            return pos;
+        }
+
+        pos += strlen(token) + 1;
+        token = strtok(NULL, " ,.!?\r\t\n");
+
+    }
+
+    return -1;
+
+}
+
+list_t* parse_file_buffered(FILE* pf, int start_line, int end_line, char* word, list_t* matches) {
+
+    char* line = NULL;
+    int wordlen = strlen(word);
+    size_t len = 0;
+    ssize_t read;
+
+    int found = -1;
+    int lineNum = start_line;
+
+    while (lineNum <= end_line && (read = getline(&line, &len, pf)) != -1) {
+
+        char sanitized[strlen(line) + 1];
+        strncpy(sanitized, line, strlen(line));
+        sanitized[strcspn(sanitized, "\r\n")] = 0;
```

Comment:
Let's set this to `'\0'` instead of `0` so that we're not abusing the type system.

Reply:
Changed